### ISSUE TITLE
raft branch: building breaking on MacOS because of  `debug_print`

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#4312]
On MacOS the current version of the `raft` branch is failing to build with the message:
```
src/clustering/table_contract/contract_metadata.cc:190:5: error: call to 'debug_print' is ambiguous
    debug_print(buf, ack.branch_history.branches.size());
    ^~~~~~~~~~~
./src/debug.hpp:29:6: note: candidate function
void debug_print(printf_buffer_t *buf, int x);
     ^
./src/debug.hpp:30:6: note: candidate function
void debug_print(printf_buffer_t *buf, uint64_t x);
     ^
```

As the error message points out this is because it can't decide which version of `debug_print` to convert to because on MacOS because `size_t` is defined as something other than a `uint64` (I got as far as `__darwin_size_t` before giving up). With @VeXocide's help I tried working around this by using a `static_cast<uint64_t>` in the offending line, but that just got me to the to the next similar problem. This would probably work, but would create a whack-a-mole problem trying to keep up with all of this class of problem for all the different platforms.

We also tried to see if we could either add a `size_t` version of `debug_print` (works on MacOS, fails on newton because of duplicate signatures) or remove the `uint64_t` version (used other places), but neither worked.

**Comment by Tryneus** [comment]
Fix is up in review 2931.

**Comment by Tryneus** [comment]
The fix has been approved and merged to `raft` in commit 8840e36.
